File: src/main/js/g0001_0100/s0007_reverse_integer/solution.js
// #Medium #Top_Interview_Questions #Math #Udemy_Integers
// #2024_12_04_Time_67_ms_(85.38%)_Space_53.1_MB_(34.57%)

/**
 * @param {number} x
 * @return {number}
 */
var reverse = function (x) {
    let rev = 0
    while (x !== 0) {
        rev = rev * 10 + (x % 10)
        x = Math.trunc(x / 10)
    }
    if (rev > Math.pow(2, 31) - 1 || rev < -Math.pow(2, 31)) {
        return 0
    }
    return rev
}

export { reverse }

File: src/main/erlang/g0001_0100/s0007_reverse_integer/Solution.erl
% #Medium #Top_Interview_Questions #Math #Udemy_Integers
% #2025_01_08_Time_244_ms_(100.00%)_Space_58.56_MB_(100.00%)

-spec reverse(X :: integer()) -> integer().
reverse(X) -> reverse(X, 0).
reverse(0, Work) ->
  case abs(Work) > 2147483647 of
    true -> reverse(0,0);
    false -> Work
  end;
reverse(X, Work) -> reverse(trunc(X/10), (Work*10) + X rem 10).
